src/core/qgsmagneticmodel.h

```diff
@@ -0,0 +1,299 @@
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     */
+    QString description() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns the date of the magnetic model, if available.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     */
+    QDateTime dateTime() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns the full file name for the magnetic model.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     */
+    QString file() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns the full directory name containing the magnetic model file.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     */
+    QString directory() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns the name of the magnetic model.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     */
+    QString name() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns a string for the last error encountered by the model.
+     */
+    QString error() const { return mError; }
+
+    /**
+     * Returns the minimum height above the ellipsoid (in meters) for which the model should be used.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     * \see maximumHeight()
+     */
+    double minimumHeight() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns the maximum height above the ellipsoid (in meters) for which the model should be used.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     * \see minimumHeight()
+     */
+    double maximumHeight() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns the minimum time (in decimal years) for which the model should be used.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     * \see maximumYear()
+     */
+    double minimumYear() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns the maximum time (in decimal years) for which the model should be used.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     * \see minimumYear()
+     */
+    double maximumYear() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns the equatorial radius of the associated ellipsoid (in meters).
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     * \see flattening()
+     */
+    double equatorialRadius() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Returns the flattening of the associated ellipsoid.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     * \see equatorialRadius()
+     */
+    double flattening() const SIP_THROW( QgsNotSupportedException );
+
```

Comment:
I realize you're just mapping the underlying library API, but wouldn't it be better to report a QgsCoordinateReferenceSystem object ?   Although we might have troubles constructing it since we only know about the ellipsoid shape, but not the datum. I assume this must be something alike WGS84 ...

Reply:
Done